scripts/bootstrap

```diff
@@ -16,15 +16,51 @@ function run() {
     fi
 }
 
+# Ask user for confirmation for action "$1" before going further
+# (skipped on CI and non-interactive shells in general)
+function confirm() {
+    if [[ -n ${CI:-} || -z ${PS1:-} ]]; then return; fi
+    read -p "$1 [y/n]" -n 1 -r
+    if [[ ! $REPLY =~ ^[Yy]$ ]]
+    then
+        exit 1
+    fi
+}
+
 rust_version=$(cat ./rust-toolchain.toml | sed -n 's/^channel[[:space:]]*=[[:space:]]"\(.*\)"/\1/p')
-echo "using rust version '$rust_version'"
+confirm "Install rustup with rust '$rust_version'?"
 
+echo "Installing rust '$rust_version'"
 # here we set the toolchain to 'none' and rustup will pick up on ./rust-toolchain.toml
 run curl --fail https://sh.rustup.rs -sSf | run sh -s -- -y --default-toolchain "none" --no-modify-path
 
 echo "looking for ic-cdk-optimizer 0.3.4"
 if [[ ! "$(command -v ic-cdk-optimizer)" || "$(ic-cdk-optimizer --version)" != "ic-cdk-optimizer 0.3.4" ]]
 then
-    echo "installing ic-cdk-optimizer 0.3.4"
-    run cargo install ic-cdk-optimizer --version 0.3.4
+    bin_dir="$HOME/.local/bin"
```

Comment:
Shouldn't this be `/usr/local/bin` on macos?

Reply:
Why particularly on macOS? in general I think it's not really a good idea to install anything globally; at least `$HOME/.local` is per-user

Reply:
I thought you wanted to use a standard directory. On my Mac I don't have the `$HOME/.local/bin` directory and also not added to the path. I.e. this adds more friction for users who want to run the script.

But given it's `bootstrap`, it might not be that important.

Reply:
It's not "standard" standard, but it's the directory python and a lot of package managers use; I'm happy to go with something else if you have suggestions (although nothing global)

Reply:
Ah ok. Yeah, I think it's ok.